### PR TITLE
feat: support error codes

### PR DESCRIPTION
**Pull request by Kristiyan-Todorov**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -24,6 +25,10 @@ function UTError(x) {
             }.bind(this));
         }
     }
+    if (!this.code) {
+        // unknown error in iso8583
+        this.code = 96;
```
Comment (kalinkrustev) [inline index.js]: Is there a way to not put iso8583 specifics in this module?

**Review by kalinkrustev — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (Kristiyan-Todorov) [reply index.js]: Depends on whether anyone suggests a better nomenclature.

**Comment by kalinkrustev** [comment]
I still hesitate to allow codes in the error objects. It is too tempting to base logic on these codes, while we have to base the logic on error.type. Codes should appear as part of mapping when errors need to be sent to external systems - like DBs and third party systems. Same error can have multiple codes, depending on the external system. So each error code should also have that information associated to it - i.e. which internal system it relates to. But if possible, we should avoid codes at all until error exits our system.

**Comment by kalinkrustev** [comment]
@stambata, can you read my comment and if you have opinion, please let us know.

**Pull request closed without merging**